am-lib/src/test/java/uk/gov/hmcts/reform/amlib/helpers/TestConstants.java

```diff
@@ -23,10 +26,25 @@
     public static final String OTHER_ACCESSOR_ID = "b";
     public static final Set<Permission> EXPLICIT_READ_CREATE_UPDATE_PERMISSIONS =
         Stream.of(CREATE, READ, UPDATE).collect(toSet());
+    public static final Set<Permission> EXPLICIT_READ_PERMISSION = Stream.of(READ).collect(toSet());
+    public static final Set<Permission> EXPLICIT_CREATE_PERMISSION = Stream.of(CREATE).collect(toSet());
+    public static final Map<JsonPointer, Set<Permission>> EMPTY_ATTRIBUTE_PERMISSIONS = new ConcurrentHashMap<>();
+    public static final Map<JsonPointer, Set<Permission>> SINGLE_ATTRIBUTE_PERMISSION = new ConcurrentHashMap<>();
```

Comment:
I do not like names like `SINGLE_ATTRIBUTE_PERMISSION` or `MULTIPLE_ATTRIBUTE_PERMISSIONS`. When I was reading tests code it was difficult to understand change without knowing what `SINGLE_ATTRIBUTE_PERMISSION` and `MULTIPLE_ATTRIBUTE_PERMISSIONS` do.

Maybe we should avoid using constants for it and every test should explicitly create new instance. For example instead of

```
ams.grantExplicitResourceAccess(createRecord(resourceId, ACCESSOR_ID, MULTIPLE_ATTRIBUTE_PERMISSIONS));
```

we could do:

```
... permissions = MapBuilder.
  .put("", EXPLICIT_READ_CREATE_UPDATE_PERMISSIONS)
  .put("/name", EXPLICIT_READ_PERMISSION)
  .build()

ams.grantExplicitResourceAccess(createRecord(resourceId, ACCESSOR_ID, permissions));
```

Constants hide too much IMO. They ate fine for things that does not matter like security classification. Permissions and attributes that are at core should be more visible.